pkg/repositories/gormimpl/artifact.go

```diff
@@ -73,3 +74,31 @@ func (h *artifactRepo) Get(ctx context.Context, in models.ArtifactKey) (models.A
 
 	return artifact, nil
 }
+
+func (h *artifactRepo) List(ctx context.Context, datasetKey models.DatasetKey, in models.ListModelsInput) ([]models.Artifact, error) {
+	timer := h.repoMetrics.ListDuration.Start(ctx)
+	defer timer.Stop()
+
+	artifacts := make([]models.Artifact, 0)
+	sourceEntity := common.Artifact
+
+	// add filter for dataset
+	datasetUUIDFilter := NewGormValueFilter(sourceEntity, common.Equal, "dataset_uuid", datasetKey.UUID)
+	in.Filters = append(in.Filters, datasetUUIDFilter)
+
+	// apply filters and joins
+	tx, err := applyListModelsInput(h.db, sourceEntity, in)
+
+	if err != nil {
+		return nil, err
+	} else if tx.Error != nil {
```

Comment:
I want to fail fast in case there were errors that happened in `applyListModelsInput` before issuing other calls to the `tx`. For example the `tx.join`s issued in that method may result in some errors and we shouldn't continue and try to make a db call if that was the case.